Replicator/IncomingRev.cc

```diff
@@ -49,13 +50,13 @@ namespace litecore::repl {
 
     // Read the 'rev' message, then parse either synchronously or asynchronously.
     // This runs on the caller's (Puller's) thread.
-    void IncomingRev::handleRev(blip::MessageIn* msg, uint64_t bodySize) {
+    void IncomingRev::handleRev(Retained<blip::MessageIn> msg, uint64_t bodySize) {
         reinitialize();
         _bodySize = bodySize;
 
         // Set up to handle the current message:
         DebugAssert(!_revMessage);
-        _revMessage         = msg;
+        _revMessage         = std::move(msg);
```

Comment:
Why move a ref counted reference?

Reply:
Because it still has memory allocated, and it is not used after here, so we move ownership. It is likely the compiler does this already as an optimisation, its just better to be sure.

Reply:
Without move, because you shared it with _revMessage, as the function returns, the ref count will be decreased and that's all. Retained only holds a pointer.

Reply:
Which is still memory on the stack. A move is always better than a copy. A copy here is calling extra code for no reason (the `Retained` class' copy constructor) when the instance that we're copying will be destroyed straight after anyway.

Reply:
This scenario is the exact type of scenario that is perfect for `std::move` (when a variable will no longer be used in one place, but will now be used in another, and it is more efficient to move ownership of the existing memory than to copy)

Reply:
I don't think move will affect the memory footprint.It merely saves a couple of counter arithmetics.

Reply:
Right, so even if it "merely saves a couple of counter arithmetics", why not perform it? It doesn't cost us anything.

Reply:
And, this code is called as part of `Retained` constructor:
```c++
    __hot void assignRef(RefCounted* &holder, RefCounted *newValue) noexcept {
        RefCounted *oldValue = holder;
        if (_usuallyTrue(newValue != oldValue)) {
            if (newValue) newValue->_retain();
            holder = newValue;
            if (oldValue) oldValue->_release();
        }
    }
```
A new pointer on the stack (`oldValue`), a pointer comparison, the arithmetic of `retain` and `release`, and copying a pointer value. All small operations, but unnecessary here, and add up on each call of `handleRev`